### PR TITLE
Some changes in reference Documentation 

### PR DESCRIPTION
**Pull request by CharmisG**
Resolved below points : 

1) This name doesn't match the name in [the documentation](https://github.com/square/react-native-square-reader-sdk/blob/master/docs/reference.md#money), Money.
2) amount should be required [per documentation](https://github.com/square/react-native-square-reader-sdk/blob/master/docs/reference.md#money).

### CODE REVIEW

File: docs/reference.md [hunk 1]
```diff
@@ -587,15 +587,15 @@ Field        | Type    | Description
...
+All `Money` objects require an `AmountMoney` and `currencyCode` but `currenyCode` is
 optional for the Reader SDK React Native plugin because `Money` objects will use
 the currency code of the currently authorized location by default.
 
 #### Example JSON
 
 ```json
 {
-  "amount": 100,
+  "AmountMoney": 100,
```
Comment (hukid) [inline docs/reference.md:598]: This is wrong. it's still "amount", the AmountMoney is the Model type, not the body.

File: src/models/CheckoutParams.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ interface CheckoutParams {
 
     note?: string;
     
-    tipSettings?:TipSettings;
+    tipSettings:TipSettings;
```
Comment (hukid) [inline src/models/CheckoutParams.ts:19]: This should be optional per this doc: https://developer.squareup.com/docs/api/reader/ios/. 

Does this throw on iOS or Android?

**Review by hukid — CHANGES_REQUESTED** [review]

File: docs/reference.md (hunk 1 above)
Comment (CharmisG) [reply docs/reference.md:598]: Then where we need to change that AmountMoney name ??

File: src/models/CheckoutParams.ts (hunk 2 above)
Comment (CharmisG) [reply src/models/CheckoutParams.ts:19]: Actually No i'm not getting any exception with this.

**Comment by hukid** [comment]
- [ ] please address this comment: https://github.com/square/react-native-square-reader-sdk/pull/148#discussion_r987183078

File: src/models/CheckoutParams.ts (hunk 2 above)
Comment (hukid) [reply src/models/CheckoutParams.ts:19]: Do you mean tipSettings can be "undefined" and no exception throw?  if so, can you please check with @caleb-harrelson in this comment https://github.com/square/react-native-square-reader-sdk/pull/148#discussion_r987182878 ?

File: docs/reference.md (hunk 1 above)
Comment (CharmisG) [reply docs/reference.md:598]: Done

**Review by hukid — APPROVED** [review]

File: src/models/CheckoutParams.ts (hunk 2 above)
Comment (caleb-harrelson) [reply src/models/CheckoutParams.ts]: > Do you mean tipSettings can be "undefined" and no exception throw? 

The changes in this PR include the changes in #198, so the answer here would be yes. 

I still maintain that the correct remedy is #199, which reverts the entire PR that “added” TypeScript in a partial implementation that does not expose the types and contains more inaccuracies than were fixed herein.

File: src/models/AmountMoney.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 interface AmountMoney {
     amount: number;
     
-    currencyCode?: string;   
+    currencyCode?: string;
+
```
Comment (hukid) [inline src/models/AmountMoney.ts:5]: Do you need this additional line?

Comment (CharmisG) [reply src/models/AmountMoney.ts:5]: Removed it.

**Review by hukid — APPROVED** [review]

**Pull request merged**